### PR TITLE
New module: tor_rate

### PR DESCRIPTION
**Pull request by fmorgner**
The tor_rate module shows the incoming and outgoing data rates of a Tor
daemon instance.

### CODE REVIEW

File: py3status/modules/tor_rate.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    control_password: The password to use for the Tor control connection
+        (default None)
+    control_port: The port on which the Tor daemon listens for control
+        connections (default 9051)
+    format: A string describing the output format for the module
+        (default "↑ {up} ↓ {down}")
+    rate_unit: The unit to use for the transfer rates
+        (default "B/s")
+    si_units: A boolean value selecting whether or not to use SI units
+        (default 0)
```
Comment (tobes) [inline py3status/modules/tor_rate.py]: `False` would be a better default

File: py3status/modules/tor_rate.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    value_format: A string describing how to format the transfer rates
+        (default "[\?min_length=12 {rate:.1f} {unit}]")
```
Comment (tobes) [inline py3status/modules/tor_rate.py]: `format_value` would be a better name.  You should document the placeholders for this format too.

File: py3status/modules/tor_rate.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+Example:
+
+```
+tor_rate {
+    cache_timeout = 10
+    format = "IN: {down} | OUT: {up}"
+    control_port = 1337
+    control_password = "TertiaryAdjunctOfUnimatrix01"
+    si_units = 1
```
Comment (tobes) [inline py3status/modules/tor_rate.py]: `si_units = True` yes we can use `0` and `1` but a boolean looks nicer.

File: py3status/modules/tor_rate.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    def __init__(self):
+        self._down = 0
+        self._handler_active = False
+        self._up = 0
+
+    def tor_rate(self, outputs, config):
+        text = ''
+        if not self._handler_active:
+            try:
+                self._register_event_handler()
```
Comment (tobes) [inline py3status/modules/tor_rate.py]: is this something we want to keep doing if it fails?  for some it feels ok eg no connection, but for authentication fail should we just give up?

**Comment by tobes** [comment]
I haven't been able to test this but it looks good to me.

@fmorgner thanks for this

**Comment by ultrabug** [comment]
Thanks a lot for your contribution @fmorgner ! :)

**Pull request merged**